src/Microsoft.Dism/Utilities.cs

```diff
@@ -0,0 +1,287 @@
+/// older/other versions/architectures, etc.)
+/// 
+/// WITH RESPECT TO DISM TOOLING:
+/// 
+/// 1.  Microsoft was calling the toolset "Windows Assessment and Installation Kit" as of Windows 7, but now calls it the "Windows Assessment and Deployment Kit" (starting with 8.0 and continuing on.)
+/// 
+/// 2.  WAIK (Windows 7) can be installed to provide Windows PE 3.0 (and DISM servicing), OR WAIK Service Pack 1 can be installed for Windows PE 3.1 (and DISM servicing), but both cannot be
+///     installed at the same time.
+///     
+/// 3.  WADK (Windows 8.0) can be installed to provide support for Windows PE 4.0 (and DISM servicing), OR WADK (Windows 8.1) can be installed for Windows PE 5.0 (and DISM servicing), but both
+///     cannot be installed at the same time.
+///     
+/// 4.  Only one version of WADK (Windows 10.x) can be installed to provide support for Windows PE 10 (and DISM servicing) at the same time.
+/// 
+/// 5.  Thus, a user could potentially have WAIK 3.0 or 3.1 installed, WADK 4.0 or 5.0 installed, and WADK 10 installed, all at the same time. In doing so, they could create and service Windows PE images
+///     from Windows 7/7 SP1, 8/8.1, and 10.
+///     
+/// 6.  Regardless of what OS you (or a user) has, it is recommended to install the latest version of the WADK (10.x as of this writing), as the latest tooling should contain support for and be able to service older images.
+///     Therefore, WADK 10.x (and it's DISM libraries) can service images of Windows 7, 8, 8.1, 10.x.
+///     
+/// LOADING A SPECIFIC DISM LIBRARY:
+/// 
+/// 1.  Use the Properties ("WADK10DISMAPIPath") to determine if said version of WADK/WAIK is installed, and if so, where the DISM API libraries are installed. 
+///     "GetLatestDismGeneration()" will search all paths and provide the latest installed version, if available, as an enumeration.
+///     
+/// 2.  Call "LoadDismGenerationLibrary()" BEFORE you call DismApi.Initialize(). This works because we use the native "LoadLibrary()" method to load the specified DISM generational library into the process space,
+///     and when you call "DismApi.Initialize()", the Marshal (via DllImport()) looks in process space first for a loaded module. Upon finding a module that matches the requested one's name, it then loads it 
+///     (and any other associated libraries) as needed.
+///     
+/// 3.  After calling "DismApi.Shutdown()", call "UnloadDismGenerationLibrary()" to remove the loaded DISM library from the process space. 
+/// 
+/// 4.  In this fashion, one could load DISM from WADK 10, unload it, load DISM from WAIK, unload it, load DISM from the System, etc.
+/// 
+/// 5.  This method of loading/unloading DISM generation libraries supports ANYCPU, X86, and AMD64, and also supports the CLR "Prefer 32-bit" setting.
+/// 
+
+using Microsoft.Win32;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Dism
+{
+    public static class Utilities

```

Comment:
I'd like this class to be named `DismUtilities`.  Do you think it makes sense to just add an overload to `DismApi.Initialize()` which does all of this rather than a different class?  It might not be intuitive to users to call another class before calling `Initialize`.  But that also changes the `DismInitialize` definition from stock.  Just a thought.  Overall this implementation looks good, I just want to know if you think we should make it easier for people to find the methods.

Reply:
I agree (and had similar thoughts), but figured I'd start with something generic so you could review and start a discussion. At first, I did overload DismInitialize(), but then reconsidered because of how cleanly it maps to the C API.

I also originally named the class "DismUtilities", but recanted because I thought it implied that it might be part of the C API as well. 

I do like how purely the library sticks to the DISM API, so I think preserving that would be more important than adding this (expanded) functionality to it (via overloading DismInitiaize(), etc.) That said, I do see the value in making it more easily accessible/discoverable. Perhaps we could expand the intellisense remarks in DismInitialize() to point users to the Utilities class (whatever it becomes), or to a FAQ or other document that expands on how to leverage some more 'advanced' features?

Reply:
@josemesona Additional thoughts? It sounds like we're basically on the same page, so I'm deferring to your judgment.

Reply:
Yeah I think I want to make the DismUtilities `internal` for now.  I think having an overload to `DismApi.Initialize()` would be best.  I can merge this and then make the class internal and do the overload if you want.